chandra_aca/tests/test_maude_decom.py

```diff
@@ -777,3 +777,61 @@ def test_imgtype_dnld(source):
     img_dnld = img["IMGTYPE"] == 3
     assert img_dnld.shape[0] > 0
     assert np.all(all_masked == img_dnld)
+
+
+def test_repeated_frames():
+    import pickle
+
+    with open(
+        os.path.join(os.path.dirname(__file__), "data", "repeat_frames.pkl"), "rb"
+    ) as fh:
+        frames = pickle.load(fh)
+    # this has repeated frames
+    packets = maude_decom.get_aca_packets(
+        start="2023:297:12:00:00", stop="2023:297:12:02:00", frames=frames
+    )
+    assert len(packets) == 40
+
+    # and removing one VCDU frame results in one ACA packet missing (eight slots)
+    frames["data"]["frames"] = (
+        frames["data"]["frames"][:7] + frames["data"]["frames"][7 + 1 :]
+    )
+    packets = maude_decom.get_aca_packets(
+        start="2023:297:12:00:00", stop="2023:297:12:02:00", frames=frames
+    )
+    assert len(packets) == 32
+
+    # the following checks the underlying function used to filter out repeated/missing frames
```

Comment:
Can you write these tests like below. This will make it much easier for me to review the tests.
```
filter_vcdu_test_cases = [
    {
        "inp": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9],
        "exp": [1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
    },
    {
        "inp": [0, 1, 3, 2, 4, 5, 6, 7, 8, 9],
        "exp": [0, 0, 0, 0, 1, 1, 1, 1, 1, 1],
    },
    {
        "inp": [0, 1, 2, 4, 5, 6, 7, 8, 9],
        "exp": [0, 0, 0, 1, 1, 1, 1, 1, 1],
    },
    ...,
]


@pytest.mark.parametrize("case", filter_vcdu_test_cases)
def test_filter_vcdu_jumps(case):
    ...
```

Reply:
Done.

Reply:
ok, I just changed that.

Reply:
I'm sorry there is some miscommunication. Perhaps you could invest just a couple of minutes (I doubt it would take you more than two minutes) to convince yourself that this checks what you requested.

The only difference is that what you wrote above gives an integer index for the resulting groups, whereas I gave a boolean mask that tells you which frame was selected.

For example, in this case:
```{
            "inp": np.array([1, 2, 3, 4, 5, 6, 6, 7, 8, 9]),
            "exp": np.array(
                [False, False, False, True, True, True, False, True, False, False]
            ),
        },
```
frame `6` is repeated, and the one selected is the first one.

Reply:
and by the way: I understood your request above as '1' and '0' being boolean, and you wanted to check which instance of a repeated frame was chosen.

Reply:
I guess I didn't explain. The entire point of what I wrote was so that the mask values would **visually line up** with the vcdu numbers. That would make it easy for me to review and make sure the output is what I would expect. 

The way you wrote it is just too much work for me as a reviewer to mentally line things up, so I just didn't bother and just 🤞  it's OK.

Also the `np.array` wrappers are not in what I wrote and they can cause the numbers to not line up visually again.

Reply:
yes, I totally did not get that what you meant was to line up visually. I understood that what you meant was to actually check which frame was being selected (the previous version of the test did not check that, and that is the change I made).

Reply:
You also did not pull the test cases out and use `pytest.mark.parametrize` as I requested. I think you understand why that is important for testing.